crypto/rsa/rsa_lib.c

```diff
@@ -259,6 +310,38 @@ void RSA_get0_factors(const RSA *r, const BIGNUM **p, const BIGNUM **q)
         *q = r->q;
 }
 
+size_t RSA_get_multi_prime_num(const RSA *r)
+{
+    int pnum;
+
+    pnum = sk_RSA_PRIME_INFO_num(r->prime_infos);
+    if (pnum <= 0)
+        pnum = 0;
+    return (size_t)pnum;
```

Comment:
Hmmm, I guess my original plan is to return the 'extra' primes number, to stay consistent with other `RSA_(get|set)_multi_prime_xxx` functions who handle the extra primes only. So do you think it's better to return the whole number of primes?

Reply:
Hmm, I guess I just find it counter-intuitive to have `RSA_get_multi_prime_num()` return 2 less than the actual number of primes - but I don't have that strong a view on it.

Reply:
Perhaps renaming it to RSA_get_multi_prime_extra_count - that way it would be clearer what it does.
Or have a RSA_get_total_prime_count() and RSA_get_multi_prime_count() as well if you want to know the difference always. Are you using this for a is-multi-prime sort of test?

Reply:
I take the name `RSA_get_multi_prime_extra_count` at current stage. And yes, this function could be used to determine if an RSA object represents a multi-prime key and it's also used to find out how much memory one needs to allocate before calling functions as `RSA_get0_multi_prime_factors`

Reply:
Out of curiosity why size_t? The value is known to be positive and limited by 16...